### ISSUE TITLE
Allow different editor for commit messages and opening files

### ISSUE THREAD

**Issue by musjj** [altsem/gitu#246]
I have my `$GIT_EDITOR` set to `code -rw`, which waits until the file is closed in VS Code. This is good for editing commit messages, but not for opening files.

It would be nice if we can configure a different editor for opening files in staged changes, untracked files, etc. Or maybe it'd be more sensible to use `$EDITOR` here instead?

**Comment by eekhof** [comment]
I just ran into a very similar problem which would also be solved by this: When using gitu from a neovim-terminal, I set my `GIT_EDITOR=nvim` and my `EDITOR=nvr` (neovim remote).

I was hoping this would me allow to open files without opening a nested neovim instance, but instead delegate them to the nvim instance that already hosts the terminal running gitu, while on the other hand letting me write my commit message in a nested neovim instance.

The latter part is because when trying to commit from nvim in the host instance called by nvr, the commit always fails. I also think using `GIT_EDITOR` and `EDITOR` in this way would be very useful. Otherwise I may need to get to the bottom of why exactly the commit fails, I hope to do that soon.
